.teamcity/common/performance-test-extensions.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package common
+
+import configurations.buildJavaHome
+import configurations.coordinatorPerformanceTestJavaHome
+import jetbrains.buildServer.configs.kotlin.v2018_2.BuildType
+
+fun BuildType.applyPerformanceTestSettings(os: Os = Os.linux, timeout: Int = 30) {
+    applyDefaultSettings(os = os, timeout = timeout)
+    artifactRules = """
+        build/report-*-performance-tests.zip => .
+    """.trimIndent()
+    detectHangingBuilds = false
+    requirements {
+        // TODO this can be removed once https://github.com/gradle/gradle-private/issues/1861 is closed
```

Comment:
https://github.com/gradle/gradle-private/issues/1861 has been closed. You can leave it here because anyway I need to resolve conflicts between https://github.com/gradle/gradle/pull/8801 and this PR, and I'll fix this comment.